app/javascript/chat/chat.jsx

```diff
@@ -768,22 +769,23 @@ export default class Chat extends Component {
     );
   };
 
-  triggerSwitchChannel = (id, slug) => {
+  triggerSwitchChannel = (id, slug, channels) => {
     const {
       chatChannels,
       isMobileDevice,
       unopenedChannelIds,
       activeChannelId,
       currentUserId,
     } = this.state;
+    const channelList = channels || chatChannels;
```

Comment:
This fixes an issue where we have the list of channels already even though Preact component state hasn't been updated yet. This is really a hack because we already have the list of channels.